cli/src/cli_util.rs

```diff
@@ -490,6 +490,7 @@ pub struct WorkspaceCommandHelper {
     may_update_working_copy: bool,
     working_copy_shared_with_git: bool,
     path_converter: RepoPathUiConverter,
+    reset_author_revset: OnceCell<Rc<RevsetExpression>>,
```

Comment:
nit: maybe it's okay to parse revset at every call (without caching)? Parsing is cheap compared to evaluation, and I don't think `.rewrite_edited_commit()` is called in a hot loop.